apps/researcher/src/lib/community.ts

```diff
@@ -0,0 +1,91 @@
+import {clerkClient, auth} from '@clerk/nextjs';
+
+type Community = {
+  id: string;
+  name: string;
+};
+
+export type Membership = {
+  id: string;
+  role: string;
+  publicUserData?: {
+    userId: string;
+    firstName: string | null;
+    lastName: string | null;
+    profileImageUrl: string;
+  } | null;
+};
+
+interface Error {
+  status: number;
+}
+
+interface CommunityResponse {
+  community: Community;
+  error: undefined;
+}
+
+interface CommunityErrorResponse {
+  community: undefined;
+  error: Error;
+}
+
+export async function getCommunity(
+  communityId: string
+): Promise<CommunityResponse | CommunityErrorResponse> {
+  let community, error;
+
+  try {
+    community = await clerkClient.organizations.getOrganization({
+      organizationId: communityId,
+    });
+  } catch (e) {
+    error = e as Error;
+    return {community: undefined, error};
```

Comment:
Would it be an option to just throw the Clerk error (and catch it in the caller's code rather than here)? Or if you don't want Clerk errors to 'leak' into the caller's code: catch it here and throw our own (instance of) `Error`?

Reply:
I was in doubt which approach to take for error handling. I will just throw the Clerk error.